lessons/dotnetcore31/durable-functions/chaining/README.md

```diff
@@ -0,0 +1,333 @@
+   > 🔎 **Observation** - The status endpoint returns not only the result but also some metadata with respect to the overall execution.
+
+7. Check the resulting entries in your Azure Storage Emulator
+   > ❔ **Question** - How many tables have been created by the Azure Functions runtime? What do they contain?
+
+## 3. Implementing a "Real-World" Scenario
+
+In this section we develop some more realistic setup for our Durable Function. Assume the following situation:
+Assume that we have the name of a GitHub repository. Now we want to find out who is the owner of this repo. In addition we want to find our some more things about the owner like the real name, the bio etc. To achieve this we must execute two calls in a sequence to the GitHub API:
+
+1. Get the information about the repository itself containing the user ID.
+2. Based on the user ID we fetch the additional information of the user.
+
+The good thing about the GitHub API is that we do not need to care about authentication and API keys. This means that there are some restrictions with respect to the allowed number of calls per minute, but that is fine for our scenario.
+
+### 3.1 Basic Setup of "Real-World" Scenario
+
+In this section we add the skeleton for the implementation.
+
+#### Steps
+
+1. Create a new function in a folder named `GitHub` via the Azure Functions Extension in VSCode:
+   1. Select `Durable Functions orchestrator` as a template.
+   2. Name the function `GitHubInfoOrchestrator`.
+   3. Delete the orchestration function body
+   4. Use `Durable.Chaining.GitHub` as namespace
+   5. Delete the `SayHello` template's function and add two new functions:
+      1. `GetRepositoryDetailsByName`
+      2. `GetUserDetailsByName`
+   6. Install the following NuGet package for a smooth interaction with the GitHub REST API
+      1. `Octokit`
+
+### 3.2 Implementation of the Orchestrator Function
+
+In this section we implement the Orchestrator Function that defines the call sequence of the Activity Functions and assures the transfer of the result of the first Activity Function to the second one.
+
+#### Steps
+
+1. Update the Orchestrator Function `GitHubInfoOrchestrator` to call the two Activity Functions. The first Activity Function is `GetRepositoryDetailsByName`. The returned information of that activity, the user ID, must be transferred to the context of the Durable Function. The second Activity Function is `GetUserDetailsById`. The returned information of the second Activity Function is returned as response by the Orchestrator Function. The Orchestrator Function code should look like this:
+
+   ```csharp
+   [FunctionName("GitHubInfoOrchestrator")]
+   public static async Task<UserDetails> RunOrchestrator(
+      [OrchestrationTrigger] IDurableOrchestrationContext context)
+   {
+      var input = context.GetInput<string>();
+
+      var userName = await context.CallActivityAsync<string>("GitHubInfoOrchestrator_GetRepositoryDetailsByName", input);
```

Comment:
Personally I never use the orchestrator name as a prefix for the activity name. I see the template does this, I don't know if this is a recommended practice but the names get rather long this way. Shall we just use the function name without the orchestrator prefix? This would be more inline with section 3.1 step 5.

Reply:
Yeah, I moved looking at the documentation but what you said makes sense to me